src/cordova/requirements.js

```diff
@@ -32,21 +32,18 @@ var knownPlatforms = require('../platforms/platforms');
  *   requirements check results for each platform.
  */
 module.exports = function check_reqs (platforms) {
-    return Q().then(function () {
-        var platforms = cordova_util.preProcessOptions(platforms).platforms; // eslint-disable-line no-use-before-define
+    return Promise.resolve().then(() => {
```

Comment:
Is there a reason to wrap this in a promise, given that the inner block is already returning `Promise.all`?

Reply:
It's a poor-man's [`Promise.try`](http://cryto.net/~joepie91/blog/2016/05/11/what-is-promise-try-and-why-does-it-matter/).

In a nutshell: It ensures that this function  __always__ returns a `Promise`. Even if anything we do would cause an error to be thrown, thanks to the wrapping it will be converted to a rejection instead of bubbling up the call stack.

In this case, suppose `preProcessOptions(platforms)` would return `undefined`. Then our accessing the result's `platforms` property would result in an error to be thrown.

Reply:
Ahh, I actually just left a comment on the cordova-fetch PR referencing `Promise.try`. Since we seem to use this pattern frequently, maybe it makes sense to depend on something like https://github.com/tc39/proposal-promise-try (although that isn't published to npm :disappointed:)?

Reply:
Yeah, sorry for doing a ELI5 here. And I agree with you regarding the lib: I just suggested the exact same thing in the cordova-fetch PR :grin: